osu.Game/Database/RealmArchiveModelImporter.cs

```diff
@@ -188,7 +188,9 @@ public async Task<ExternalEditOperation<TModel>> BeginExternalEditing(TModel mod
 
             Directory.CreateDirectory(mountedPath);
 
-            foreach (var realmFile in model.Files)
+            // Detach files from the model to avoid realm contention when copying to the external location.
+            // This is safe as we are not modifying the model in any way.
+            foreach (var realmFile in model.Files.Detach())
```

Comment:
As  the comment states, I ran into an issue where detaching the skin doesn't actually end up detaching the files, resulting in a `Realms.Exceptions.RealmException: Realm accessed from incorrect thread.` exception.

I couldn't find a way to fix it apart from this. Fortunately, we only access the realm files here to mount them, so we aren't modifying the files in a way that requires them to be attached.

Reply:
Two questions:

- Where / when is the "skin" being "detached"?
- Does detach even... _work_ for `SkinInfo`? It sure isn't _registered_ in the automapper detach machinery.

https://github.com/ppy/osu/blob/74675c8c7858416a241e29804ba88ca1c1f5e476/osu.Game/Database/RealmObjectExtensions.cs#L171-L179

Maybe adding `SkinInfo` in there just fixes this. I dunno.

Reply:
The skin is being detached [here](https://github.com/smallketchup82/osu/blob/15c7a12174687d5bcb9d058bcbb72a35e3a42b35/osu.Game/Overlays/SkinEditor/SkinEditor.cs#L283)

`CurrentSkin.Value.SkinInfo` in `SkinEditor` returns a `Live<SkinInfo>`. My original idea for fixing the issue with "realm being accessed from an incorrect thread" was to detach the `Live<SkinInfo>` so that we're only working with a `SkinInfo`. Though, simply stopping here (without detaching `Files`) doesn't work and results in the thread error. Adding in the `.Detach()` fixes it.

I got here via trial and error so I don't know for certain. But my best guess is that it likely has to do with `RealmNamedFileUsage` being in the automapper detach machinery in which you specified. Maybe it's only detaching the `Files` that ends up fixing the issue, not necessarily the detaching of the `Live<SkinInfo>`.

At any rate, I guess detaching the `Live<SkinInfo>` serves only to convert from `Live<SkinInfo>` to `SkinInfo`. But a quick test without detaching the `SkinInfo` seems to work fine. So maybe that's unnecessary?